ESPHome on aarch64/arm64v8

**Operating environment/Installation (Hass.io/Docker/pip/etc.):**

* ArchLinuxArm 64 bit on a Raspberry Pi 3 B+
* Docker version 18.09.3-ce, build 774a1f4eee
* Docker image esphome/esphome-aarch64 in Version 1.11.2

**ESP (ESP32/ESP8266, Board/Sonoff):**

* Wemos D1 Mini

**Affected component:**

Probably none.

**Description of problem:**

Using the docker image (as stated above) no firmware can be compiled. It seems that the toolchain has some issues. Two sections below you'll find the output of the (web-) console by trying a clean build.

**Problem-relevant YAML-configuration entries:**

Probably none.

**Traceback (if applicable):**

```
INFO Reading configuration...
INFO Generating C++ source...
INFO Compiling app...
INFO Running:  platformio run -d /config/wemos_test
Processing wemos_test (platform: espressif8266@1.8.0; framework: arduino; board: d1_mini)
--------------------------------------------------------------------------------
LibraryManager: Installing id=305 @ 1.1.3
Downloading...
Unpacking  [####################################]  100%
LibraryManager: Installing v1.11.2
Downloading...
Unpacking  [####################################]  100%          
LibraryManager: Installing id=3837 @ 5.13.3
Downloading...
Unpacking  [####################################]  100%
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif8266/d1_mini.html
PLATFORM: Espressif 8266 > WeMos D1 R2 & mini
HARDWARE: ESP8266 80MHz 80KB RAM (4MB Flash)
Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF MODES: FINDER(chain) COMPATIBILITY(soft)
Collected 27 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <ESP8266WiFi> 1.0
|-- <ESP8266mDNS>
|   |-- <ESP8266WiFi> 1.0
|-- <ESPAsyncTCP> 1.1.3
|   |-- <ESP8266WiFi> 1.0
|-- <Hash> 1.0
|-- <esphome-core> 1.11.2
|   |-- <ArduinoJson-esphomelib> 5.13.3
|   |-- <ESP8266WiFi> 1.0
|   |-- <ESP8266mDNS>
|   |   |-- <ESP8266WiFi> 1.0
|   |-- <ESPAsyncTCP> 1.1.3
|   |   |-- <ESP8266WiFi> 1.0
Compiling .pioenvs/wemos_test/src/main.cpp.o
Generating LD script .pioenvs/wemos_test/ld/eagle.app.v6.common.ld
Compiling .pioenvs/wemos_test/lib4d9/ESP8266WiFi/BearSSLHelpers.cpp.o
Compiling .pioenvs/wemos_test/lib4d9/ESP8266WiFi/CertStoreBearSSL.cpp.o
sh: 1: xtensa-lx106-elf-g++: Exec format error
*** [.pioenvs/wemos_test/src/main.cpp.o] Error 2
sh: 1: xtensa-lx106-elf-gcc: Exec format error
*** [.pioenvs/wemos_test/ld/eagle.app.v6.common.ld] Error 2
sh: 1: xtensa-lx106-elf-g++: Exec format error
*** [.pioenvs/wemos_test/lib4d9/ESP8266WiFi/CertStoreBearSSL.cpp.o] Error 2
sh: 1: xtensa-lx106-elf-g++: Exec format error
*** [.pioenvs/wemos_test/lib4d9/ESP8266WiFi/BearSSLHelpers.cpp.o] Error 2
========================== [ERROR] Took 26.75 seconds ==========================
```

**Additional information and things you've tried:**

A clean build/compilation on AMD64 plattform and upload via USB works. After that, the Wemos D1 mini is recognized by the aarch64/arm64v8 docker image.

Even with the docker images tagged with 1.11.0 and 1.11.1 I wasn't able to compile a firmware.

<!-- LEAVE THIS LINE AS-IS AND DON'T DELETE IT, OTHERWISE THE ISSUE WILL BE CLOSED AUTOMATICALLY. -->

Hi,

I tried to compile a little test firmware using version 1.13.0b3 and I obtained the same error output.

```
INFO Reading configuration...
INFO Generating C++ source...
INFO Compiling app...
INFO Running:  platformio run -d /config/wemos_test
Processing wemos_test (framework: arduino; platform: espressif8266@1.8.0; board: d1_mini)
--------------------------------------------------------------------------------
LibraryManager: Installing id=305 @ 1.2.0
Downloading...
Unpacking  [####################################]  100%
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif8266/d1_mini.html
PLATFORM: Espressif 8266 > WeMos D1 R2 & mini
HARDWARE: ESP8266 80MHz 80KB RAM (4MB Flash)
Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF MODES: FINDER(chain) COMPATIBILITY(soft)
Collected 27 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <ESP8266WiFi> 1.0
|-- <ESP8266mDNS>
|   |-- <ESP8266WiFi> 1.0
|-- <ESPAsyncTCP> 1.2.0
|   |-- <ESP8266WiFi> 1.0
Compiling .pioenvs/wemos_test/src/esphome/components/api/api_message.cpp.o
Compiling .pioenvs/wemos_test/src/esphome/components/api/api_server.cpp.o
Compiling .pioenvs/wemos_test/src/esphome/components/api/basic_messages.cpp.o
Compiling .pioenvs/wemos_test/src/esphome/components/api/command_messages.cpp.o
sh: 1: xtensa-lx106-elf-g++: Exec format error
*** [.pioenvs/wemos_test/src/esphome/components/api/api_message.cpp.o] Error 2
sh: 1: xtensa-lx106-elf-g++: Exec format error
*** [.pioenvs/wemos_test/src/esphome/components/api/command_messages.cpp.o] Error 2
sh: 1: xtensa-lx106-elf-g++: Exec format error
*** [.pioenvs/wemos_test/src/esphome/components/api/api_server.cpp.o] Error 2
sh: 1: xtensa-lx106-elf-g++: Exec format error
*** [.pioenvs/wemos_test/src/esphome/components/api/basic_messages.cpp.o] Error 2
========================== [ERROR] Took 11.98 seconds ==========================
```

I'm using the image `esphome/esphome-aarch64:1.13.0b3`. What is the difference between this image and the hassio ones?

Thanks in advance!

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -3,7 +3,7 @@
 variables:
   DOCKER_DRIVER: overlay2
   DOCKER_HOST: tcp://docker:2375/
-  BASE_VERSION: '1.7.0'
+  BASE_VERSION: '1.8.0'
   TZ: UTC
 
 stages:

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -1,4 +1,4 @@
-ARG BUILD_FROM=esphome/esphome-base-amd64:1.7.0
+ARG BUILD_FROM=esphome/esphome-base-amd64:1.8.0
 FROM ${BUILD_FROM}
 
 COPY . .

--- a/docker/Dockerfile.lint
+++ b/docker/Dockerfile.lint
@@ -1,4 +1,4 @@
-FROM esphome/esphome-base-amd64:1.7.0
+FROM esphome/esphome-base-amd64:1.8.0
 
 RUN \
     apt-get update \

